github/messages.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file provides functions for validating payloads from GitHub Webhooks.
+// GitHub docs: https://developer.github.com/webhooks/securing/#validating-payloads-from-github
+
+package github
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"hash"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const (
+	// sha1Prefix is the prefix used by Github before the HMAC hexdigest.
+	sha1Prefix = "sha1"
+	// sha256Prefix and sha512Prefix are provided for future compatibility.
+	sha256Prefix = "sha256"
+	sha512Prefix = "sha512"
+	// eventHeader is the Github header key used to pass the event type.
+	eventHeader = "X-Github-Event"
```

Comment:
@parkr - It turns out that canonical headers don't work that way. You can verify this yourself using `curl` or `bat`.
Here is an example response I just got from the GitHub API:

```
X-Github-Request-Id : 68840050:7F91:291081F8:5755B423
X-Github-Media-Type : github.v3
```

I tried searching for the `RFC` that defines this but came up empty.  I'm sure it is out there somewhere.

As for the comments correcting the brand name spelling... yes, that could all be cleaned up by someone.